src/client.js

```diff
@@ -131,6 +149,20 @@ let pushClient = {
     navigator.serviceWorker.removeEventListener('message', messageHandler);
   },
 
+  async getSubscription() {
+    if (!this.supported()) {
+      return;
```

Comment:
This first time seeing async usage so this may be a stupid Q.

Would it make sense to throw an error here to be handled in the catch event of the promise?

Returning nothing seems like it would carry on as if this was an expected flow.